Automattic/wp-calypso
# Rename user-mention-suggestions block to user-mentions

Renames `user-mention-suggestions` to `user-mentions`, and duplicates the suggestion list and suggestion components from the TinyMCE mentions plugin (neither are in use yet).

Breaking this out from https://github.com/Automattic/wp-calypso/pull/24287 to make that PR easier to follow.

### To test

Visit http://calypso.localhost:3000/devdocs/blocks/user-mentions and ensure that the page loads as below without errors.

<img width="1131" alt="screen shot 2018-04-24 at 1 37 54 pm" src="https://user-images.githubusercontent.com/17325/39161436-c44ccfba-47c4-11e8-90c9-fcbca749c406.png">
